kong/db/migrations/core/005_120_to_130.lua

```diff
@@ -1,3 +1,139 @@
+    return tostring(val)
+  elseif t_val == "string" then
+    return "'" .. tostring((val:gsub("'", "''"))) .. "'"
+  elseif t_val == "boolean" then
+    return val and "TRUE" or "FALSE"
+  end
+  error("don't know how to escape value: " .. tostring(val))
+end
+
+
+local function add_upstream_algorithm_to_postgress(connector)
+  assert(connector:connect_migrations())
+
+  local select_query = [[
+    SELECT * FROM upstreams;
+  ]]
+
+  for row, err in connector:iterate(select_query) do
+    local algorithm
+    if row.hash_on == "none" then
+      algorithm = escape_literal("round-robin")
+    else
+      algorithm = escape_literal("consistent")
+    end
+
+    local id = escape_literal(row.id)
+    local created_at = escape_literal(row.created_at)
+    local name = escape_literal(row.name or ngx.null)
+    local hash_on = escape_literal(row.hash_on or ngx.null)
+    local hash_fallback = escape_literal(row.hash_fallback or ngx.null)
+    local hash_on_header = escape_literal(row.hash_on_header or ngx.null)
+    local hash_fallback_header = escape_literal(row.hash_fallback_header or ngx.null)
+    local hash_on_cookie = escape_literal(row.hash_on_cookie or ngx.null)
+    local hash_on_cookie_path = escape_literal(row.hash_on_cookie_path or ngx.null)
+    local slots = escape_literal(row.slots or ngx.null)
+    local tags = escape_literal(row.tags or ngx.null)
+    local healthchecks_json, err = cjson.encode(row.healthchecks)
+    if err then
+      return err
+    end
+    local healthchecks = escape_literal(healthchecks_json or ngx.null)
+
+    local delete_query = string.format([[
+      DELETE FROM upstreams WHERE id = %s
+    ]], id)
+    local _, err = connector:query(delete_query)
+    if err then
+      return err
+    end
+
+    local insert_query = string.format([[
+      INSERT INTO upstreams(id, created_at, name, hash_on, hash_fallback,
+        hash_on_header, hash_fallback_header, hash_on_cookie,
+        hash_on_cookie_path, slots, healthchecks, tags, algorithm)
+      VALUES(%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+    ]], id, created_at, name, hash_on, hash_fallback, hash_on_header,
+      hash_fallback_header, hash_on_cookie, hash_on_cookie_path, slots,
+      healthchecks, tags, algorithm)
+    local _, err = connector:query(insert_query)
+    if err then
+      print(err)
+      return err
+    end
+
+  end
+end
+
+
+local function add_upstream_algorithm_to_cassandra(connector)
+  assert(connector:connect_migrations())
+
+  local select_query = [[
+    SELECT * FROM upstreams;
+  ]]
+  local insert_query = [[
+    INSERT INTO upstreams(id, created_at, name, hash_on, hash_fallback,
+      hash_on_header, hash_fallback_header, hash_on_cookie,
+      hash_on_cookie_path, slots, healthchecks, tags, algorithm)
+    VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+  ]]
+  local delete_query = [[
+      DELETE FROM upstreams WHERE id = ?
+  ]]
+
+  local rows, err = connector:query(select_query)
```

Comment:
even if you do `SELECT *`, this does not guarantee that all rows are returned in Cassandra. You have to use `iterate` (which returns pages) and then iterate the returned `rows` in an inner loop.

Reply:
(see `plugins/acl/migrations/001_14_to_15.lua` for an example)